source/common/network/utility.cc

```diff
@@ -524,9 +525,11 @@ void passPayloadToProcessor(uint64_t bytes_read, Buffer::RawSlice& slice,
                             Buffer::InstancePtr buffer, Address::InstanceConstSharedPtr peer_addess,
                             Address::InstanceConstSharedPtr local_address,
                             UdpPacketProcessor& udp_packet_processor, MonotonicTime receive_time) {
-  // Adjust used memory length.
-  slice.len_ = std::min(slice.len_, static_cast<size_t>(bytes_read));
-  buffer->commit(&slice, 1);
+  if (slice.mem_ != nullptr) {
```

Comment:
GRO code path uses move(), so commit() is not needed. In fact, Buffer::OwnedImpl should have complained about out-of-sequence commit().
How about moving commit() call to the caller, wherever reserve() is called?

Reply:
Yes, initially it complained about out-of-sequence commit(). 
That is the reason why I am sending a slice with (slice.mem_ == nullptr) in case of gro, to avoid the extra commit here!

I also thought about moving the commit() to reserve(), but that required adding redundant code at 4 different places. Hence, I decided against it!

Reply:
Since it's no longer consistent at different call sites, I prefer to move commit() to the caller instead. It is easier to read.

Reply:
Sure, thanks for the quick reply! Moving commits to where the reserves are!